pgmpy/tests/test_base/test_MAG.py

```diff
@@ -161,3 +163,43 @@ def test_graph_properties(self, mag):
         assert mag.has_edge("C", "D")
         assert mag.has_edge("A", "D")
         assert mag.has_edge("B", "C")
+
+    def test_from_dagitty(self):
+        model_str = "mag { E [latent] A [e] J [o] {B, E} -> A; A -- J ; A -- M}"
+        model_from_str = MAG.from_dagitty(model_str)
+        with open("test_model.dagitty", "w") as f:
+            f.write(model_str)
+        model_from_file = MAG.from_dagitty(filename="test_model.dagitty")
+        os.remove("test_model.dagitty")
+
+        expected_edges = {("A", "B,"), ("A", "E"), ("A", "J"), ("A", "M")}
+        expected_roles = {"outcome": ["J"], "latents": ["E"], "exposure": ["A"]}
+
+        # Handles order of nodes
+        def normalize_edge(edge):
+            return tuple(sorted(edge))
```

Comment:
I think this is incorrect. For a directed edge, it can potentially flip the direction.